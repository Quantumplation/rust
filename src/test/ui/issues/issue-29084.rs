macro_rules! foo {
    ($d:expr) => {{
        fn bar(d: u8) { }
        bar(&mut $d);
        //~^ ERROR arguments to this function are incorrect
        //~| expected `u8`, found `&mut u8`
    }}
}

fn main() {
    foo!(0u8);
    //~^ in this expansion of foo!
}
