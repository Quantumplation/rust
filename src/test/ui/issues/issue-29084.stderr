error[E0308]: arguments to this function are incorrect
  --> $DIR/issue-29084.rs:4:9
   |
LL |         bar(&mut $d);
   |         ^^^^-------^
   |             |
   |             expected `u8`, found `&mut u8`
...
LL |     foo!(0u8);
   |     ---------- in this macro invocation
   |
note: function defined here
  --> $DIR/issue-29084.rs:3:12
   |
LL |         fn bar(d: u8) { }
   |            ^^^ -----
...
LL |     foo!(0u8);
   |     ---------- in this macro invocation
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
help: provide an argument of the correct type
   |
LL |         bar({u8});
   |         ^^^^^^^^^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0308`.
